sdk/core/Azure.Core.TestFramework/src/UseSyncMethodsInterceptor.cs

```diff
@@ -89,6 +89,13 @@ public void Intercept(IInvocation invocation)
                     Type[] modelType = returnType.GenericTypeArguments;
                     Type wrapperType = typeof(SyncPageableWrapper<>).MakeGenericType(modelType);
 
+                    Type resultType = result.GetType();
```

Comment:
Let's add more tests that cover your cases to https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/tests/ClientTestBaseTests.cs#L131